js/parts-more/WaterfallSeries.js

```diff
@@ -41,14 +44,55 @@ addEvent(Chart, 'beforeRedraw', function () {
         if (series[i].options.stacking) {
             axes.forEach(function (axis) {
                 if (!axis.isXAxis) {
-                    axis.waterfallStacks = {};
+                    axis.waterfallStacks.changed = true;
                 }
             });
             i = 0;
         }
     }
 });
 
+addEvent(Axis, 'afterRender', function () {
+    var stackLabelOptions = this.options.stackLabels;
+
+    if (stackLabelOptions && stackLabelOptions.enabled &&
```

Comment:
add stackLabelOptions as default (enabled: false). Then you will avoid the checking if it exists.